sdks/java/io/cassandra/src/test/java/org/apache/beam/sdk/io/cassandra/CassandraIOTest.java

```diff
@@ -0,0 +1,234 @@
+import org.apache.beam.sdk.testing.TestPipeline;
+import org.apache.beam.sdk.transforms.Count;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.apache.beam.sdk.transforms.SimpleFunction;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+
+import org.cassandraunit.utils.EmbeddedCassandraServerHelper;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Tests of {@link CassandraIO}.
+ */
+@RunWith(JUnit4.class)
+public class CassandraIOTest implements Serializable {
+
+  private static final String CASSANDRA_KEYSPACE = "beam_ks";
+  private static final String CASSANDRA_HOST = "127.0.0.1";
+  private static final String CASSANDRA_TABLE = "beam";
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(CassandraIOTest.class);
+
+  private transient Cluster cluster;
+  private transient Session session;
+
+  @Before
+  public void startCassandra() throws Exception {
+    EmbeddedCassandraServerHelper.startEmbeddedCassandra("/cassandra.yaml",
+        "target/cassandra", 30000);
+
+    cluster = Cluster.builder().addContactPoint(CASSANDRA_HOST).withClusterName("beam").build();
+    session = cluster.connect();
+
+    try {
+      LOGGER.info("Creating the Cassandra keyspace");
+      session.execute("CREATE KEYSPACE " + CASSANDRA_KEYSPACE + " WITH REPLICATION = "
+          + "{'class':'SimpleStrategy', 'replication_factor':3};");
+      LOGGER.info(CASSANDRA_KEYSPACE + " keyspace created");
+    } catch (Exception e) {
+      // nothing to do
+    }
+
+    LOGGER.info("Use the Cassandra keyspace");
+    session.execute("USE " + CASSANDRA_KEYSPACE);
+
+    try {
+      LOGGER.info("Create Cassandra tables");
+      session.execute("CREATE TABLE person(person_id int, person_name text, PRIMARY KEY"
+          + "(person_id));");
+      session.execute("CREATE TABLE scientist(person_id int, person_name text, PRIMARY KEY"
+          + "(person_id));");
+    } catch (Exception e) {
+      // nothing to do
+    }
+
+    LOGGER.info("Insert records");
+    String[] scientists = {"Einstein", "Darwin", "Copernicus", "Pasteur", "Curie", "Faraday",
+        "Newton", "Bohr", "Galilei", "Maxwell"};
+    for (int i = 0; i < 1000; i++) {
+      int index = i % scientists.length;
+      session.execute("INSERT INTO scientist(person_id, person_name) values(" + i + ", '"
+          + scientists[index] + "');");
+    }
+  }
+
+  @After
+  public void stopCassandra() throws Exception {
+    EmbeddedCassandraServerHelper.cleanEmbeddedCassandra();
+  }
+
+  @Test
+  @Category(NeedsRunner.class)
+  public void testRead() throws Exception {
+    Pipeline pipeline = TestPipeline.create();
```

Comment:
Using `@Rule` for `TestPipeline` causes:

```
java.io.NotSerializableException: org.apache.beam.sdk.testing.TestPipeline
```
when I do `MapElements.via(new SimpleFunction())`.

I gonna investigate.